setup.py

```diff
@@ -1,11 +1,17 @@
 import os
+import subprocess
+import sys
 
 from setuptools import find_packages, setup
 
 
 def read(fname):
     return open(os.path.join(os.path.dirname(__file__), fname)).read()
 
+
+if 'sdist' in sys.argv[1:]:
+    subprocess.run(["django-admin", "compilemessages"])
```

Comment:
```suggestion
    subprocess.run(["django-admin", "compilemessages"], cwd="linkcheck")
```
I would suggest to execute it only in the linkcheck subdirectory to avoid compiling all locate lines in the virtual environment...